ansible/cloud_providers/ec2_ssh_config_setup.yml

```diff
@@ -37,18 +36,51 @@
     tags:
       - bastion_proxy_config_main
 
-  - name: Add all hosts to workdir ssh config file
+  - name: Add bastion proxy config (ip address) to workdir ssh config file
     blockinfile:
       dest: "{{ ANSIBLE_REPO_PATH }}/workdir/{{ env_type }}_{{ guid }}_ssh_conf"
-      marker: "##### {mark} ADDED Node Proxy Config  {{ item }} {{ env_type }}-{{ guid }} ######"
+      marker: "##### {mark} ADDED BASTION PROXY HOST IP {{ env_type }}-{{ guid }} ######"
+      content: |
+          Host {{ bastion_ip }}
+            Hostname {{ bastion_ip }}
+            IdentityFile ~/.ssh/{{ key_name }}.pem
+            IdentitiesOnly yes
+            User {{ remote_user }}
+            ControlMaster auto
+            ControlPath /tmp/%h-%r
+            ControlPersist 5m
+            StrictHostKeyChecking no
+    tags:
+      - bastion_proxy_config_main
+
+  - name: Add all hosts (hostname) to workdir ssh config file
+    blockinfile:
+      dest: "{{ ANSIBLE_REPO_PATH }}/workdir/{{ env_type }}_{{ guid }}_ssh_conf"
+      marker: "##### {mark} ADDED Node Proxy Config IP {{ item }} {{ env_type }}-{{ guid }} ######"
+      block: |
+          Host {{ hostvars[item].ec2_ip_address }}
+            Hostname {{ hostvars[item].ec2_ip_address }}
+            User {{ remote_user }}
+            IdentityFile ~/.ssh/{{ key_name }}.pem
+            ProxyCommand ssh -F {{ inventory_dir }}/../workdir/{{ env_type }}_{{ guid }}_ssh_conf {{ remote_user }}@{{ bastion_hostname }} -W %h:%p
+            StrictHostKeyChecking no
+    with_items:
+      - "{{ groups[('tag_' ~ env_type ~ '_' ~ guid ~ '_proxied_host_true') | replace('-', '_')] }}"
+    tags:
+      - bastion_proxy_config_hosts
+
+  - name: Add all hosts (ip address) to workdir ssh config file
+    blockinfile:
+      dest: "{{ ANSIBLE_REPO_PATH }}/workdir/{{ env_type }}_{{ guid }}_ssh_conf"
+      marker: "##### {mark} ADDED Node Proxy Config Hostname {{ item }} {{ env_type }}-{{ guid }} ######"
       block: |
-          Host {{ item }}
-            Hostname {{ item }}
+          Host {{ hostvars[item].ec2_public_dns_name }}
+            Hostname {{ hostvars[item].ec2_public_dns_name }}
             User {{ remote_user }}
             IdentityFile ~/.ssh/{{ key_name }}.pem
-            ProxyCommand ssh {{ remote_user }}@{{ bastion_hostname }} -W %h:%p
+            ProxyCommand ssh -F {{ inventory_dir }}/../workdir/{{ env_type }}_{{ guid }}_ssh_conf {{ remote_user }}@{{ bastion_hostname }} -W %h:%p
```

Comment:
FYI, i tried this nested `-F ssh_conf` inside ssh_conf,  it's acting like a fork bomb on my laptop